docs/data-modeling.md

```diff
@@ -0,0 +1,66 @@
+
+![ScalarDB data model.](images/scalardb_data_model.png)
+
+### Objects in ScalarDB
+The ScalarDB data model has several objects.
+
+#### Namespace
+A namespace is a collection of tables analogous to an SQL namespace or database. Typically, each application creates all its tables in one namespace.
+
+#### Table
+A table is a collection of data. A namespace most often contains one or more tables, each identified by a name. Each table includes a set of partitions.
+
+#### Partition
+A partition is a collection of records and a unit of (logical) distribution to nodes. Therefore, records within the same partition are placed in the same logical location.
+
+#### Record / Row
+A record or row is a set of columns that is uniquely identifiable among all of the other records.
+
+#### Column
+A column is a fundamental data element and does not need to be broken down any further. Each record is composed of one or more columns. Each column has a data type. For details about the data type, refer to [Data-type mapping between ScalarDB and other databases](https://scalardb.scalar-labs.com/docs/latest/schema-loader#data-type-mapping-between-scalardb-and-other-databases).
+
+#### Index
+An index is a copy of the records in a single table, sorted by a column. ScalarDB queries use indexes to find data more efficiently in a table, given the values of a particular column. 
+
+
+### How to locate records
+This section discusses how to locate records from a table.
+
+#### Primary key
+A primary key uniquely identifies each record; no two records can have the same primary key. Therefore, you can locate a record by specifying a primary key. A primary key comprises a set of partition keys and clustering keys.
+
+#### Partition key
+A partition key uniquely identifies a partition. A partition key comprises a set of columns, which are called partition key columns. When you specify only a partition key, you can get a set of records that belong to the partition.
+
+#### Clustering key
+A clustering key uniquely identifies a record within a partition. It comprises a set of columns called clustering key columns. When you want to specify a clustering key, you should specify a partition key for efficient lookups. When you specify a clustering key without a partition key,
+you end up scanning all the partitions if you don’t have proper indexes. Scanning all the partitions is very time-consuming, so please do it only when you know what you are doing.
+
+Records within a partition are assumed to be sorted by clustering key columns. Therefore, you can specify a part of clustering key columns in the defined order to narrow down the results to be returned. 
+
+#### Index key
+An index key identifies records by looking up the key in indexes.
+
+## How to design your database schemas
```

Comment:
This might be out of the scope, but we could add some tips to avoid the limitation of SclarDB specification. For instance, how should we create an index for multiple columns? Currently, we do not support it, but users can achieve a similar thing by creating an actual table that has links to the desired table.